package.json

```
@@ -54,6 +54,8 @@
     "@angular/cli": "~14.0.0",
     "@angular/compiler-cli": "~14.0.0",
     "@angular/language-service": "~14.0.0",
+    "@commitlint/cli": "^17.0.3",
+    "@commitlint/config-conventional": "^17.0.3",
```

suggestion: seems like the Analog commit message guidelines more closely resemble [@commitlint/config-angular](https://github.com/conventional-changelog/commitlint/tree/master/@commitlint/config-angular).

```suggestion
    "@commitlint/config-angular": "^17.0.3",
```

---

@LayZeeDK It does closely resemble but there are a few minor differences as well. Should we update our commit guidelines to match exactly what Angular has? Or we can extend the config and customize it according to our own guidelines.

---

It doesn't need to match Angular strictly in my opinion. As long as the general format is correct, we can tweak as needed

---

All right, are you the developer to configure the rules to match Analog's guidelines, @umairhm? 🙂

---

@brandonroberts cool! 
@LayZeeDK yes, I'll give it a try 😊

---

@LayZeeDK @brandonroberts I cross-matched Analog's commit message guidelines with the config from `@commitlint/config-conventional` package. The only thing I found different is that scopes should be defined in our config. I will add the following 2 scopes in the override config, the rest all seem fine.

---

@LayZeeDK @brandonroberts I've updated the `commitlint.config.js` with the required overrides. You can run tests in the terminal like this:

```
echo "chore(scope): subject of the commit message

The body of the commit message.

Footer goes here" | commitlint -V
```